### ISSUE TITLE
add Windows (MinGW) support

### ISSUE THREAD

**Issue by hhrhhr** [dbdexter-dev/meteor_decode#3]
On Windows, by default, all files open in text mode, while POSIX-compatible operating systems open in binary. Here is a small patch that does not affect the build, but allows you to use a decoder in Windows (if you use MinGW-w64).
```
--- meteor_decode-ori/src/main.c	2019-08-14 04:59:52.000000000 +0300
+++ meteor_decode-master/src/main.c	2019-08-14 16:39:46.000000000 +0300
@@ -120,14 +120,14 @@
 	ch[2] = channel_init(apid_list[2]);
 
 	/* Open/create the output PNG */
-	if (!(out_fd = fopen(out_fname, "w"))) {
+	if (!(out_fd = fopen(out_fname, "wb"))) {
 		fatal("Could not create/open output file");
 	}
 
 	if (write_statfile) {
 		stat_fname = safealloc(strlen(out_fname) + 5 + 1);
 		sprintf(stat_fname, "%s.stat", out_fname);
-		if (!(stat_fd = fopen(stat_fname, "w"))) {
+		if (!(stat_fd = fopen(stat_fname, "wb"))) {
 			fatal("Could not create/open stat file");
 		}
 	} 

--- meteor_decode-ori/src/file.c	2019-08-14 04:59:52.000000000 +0300
+++ meteor_decode-master/src/file.c	2019-08-14 16:12:02.000000000 +0300
@@ -28,7 +28,7 @@
 	}
 
 	backend->bps = bps;
-	if(!(backend->fd = fopen(path, "r"))) {
+	if(!(backend->fd = fopen(path, "rb"))) {
 		fatal("Could not find specified file");
 		/* Not reached */
 		return NULL; 

```

**Comment by dbdexter-dev** [comment]
Fixed in commit 6ab0453. Since the stat file is just 3 lines of text, there shouldn't be any need to open it in binary mode, correct?

**Comment by hhrhhr** [comment]
> correct?
no :)

![rn_issue](https://user-images.githubusercontent.com/966048/63046762-c98bac00-bedb-11e9-8359-9099516fac5a.png)

**Comment by dbdexter-dev** [comment]
Alright, corrected in d1a6000. Thanks for the help!
